### ISSUE TITLE
ASK Dialog simulator is reporting Display as a supportedInterface, which triggers unsupported APL Directives

### ISSUE THREAD

**Issue by btburton42** [alexa/ask-cli#277]
## I'm submitting a...

<pre><code>
[X] Regression (a behavior that used to work and stopped working in a new release)
[] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

As of August 4, 2020, the `ask dialog` CLI simulator started invoking skills with this supportedInterfaces object:

```
"supportedInterfaces": {
  "Display": {
    "templateVersion": "1.0",
    "markupVersion": "1.0"
  }
}
```

This causes our skill to respond with an APL Directive, which the CLI simulator does not support.

## Expected Behavior

I would expect supportedInterfaces to be an empty object as it is with normal headless devices.

## Current Behavior

Currently, all our simulation tests are failing because we are sending an unsupported APL directive to the device, which results in Alexa rendering "There was a problem with the requested skill's response" in the CLI and sending an Invalid Directive card to alexa.amazon.com

```
Invalid Directive
Request Identifier: null
The device does not support Alexa.Presentation.APL directives
```

For more details, please examine the results of this sessionId: "amzn1.echo-api.session.24c7760b-a953-4e8c-b071-a59cbec73727"

**CLI Snapshot**

## Steps to Reproduce (for bugs)

1) Create a replay file
2) Run a simulation with a skill that supports APL directives
3) Notice that the outputSpeech response is correct but the content.caption says, "There was a problem with the requested skill's response"

## Your Environment and Context

* ask-cli version: 2.14.0

**Comment by RonWang** [comment]
Hi @btburton42 we are looking into your issue right now. In the meanwhile, could you try to use the web console to test your skill, and see if same issue would happen? Thanks in advance~

**Comment by btburton42** [comment]
@RonWang is there a way to properly simulate a headless device in the web console? From what I can tell, it only supports devices with screens, which is a bit of a disappointment. As I don't own a dot, I used https://echosim.io which responded as expected with skill content.

**Comment by kakhaUrigashvili** [comment]
@btburton42 I see that 
```
"supportedInterfaces": {
  "Display": {
    "templateVersion": "1.0",
    "markupVersion": "1.0"
  }
}
```
is part of the Skill Invocation JSON Input displayed in the Alexa Web Console. 
SMAPI API and ask dialog only support text utterance as input

https://developer.amazon.com/en-US/docs/alexa/smapi/skill-simulation-api.html

I am confused what is the regression on the ask cli side. What are the steps to reproduce? If it is a regression, have you tested that an older version of ask cli works as you expect?

**Comment by btburton42** [comment]
@kakhaUrigashvili I'm confused as well. When I revert to previous versions of the ask-cli I am also seeing that supportedInterfaces.Display object. (So, perhaps this could be better described as a bug, since it is present in all versions.)

We started seeing this in our daily test run on Aug. 4. It was not present on Aug. 3. Everything up to that point was working as expected. As your team is my only point of contact into issues with the simulator, I logged the issue here.

If you have suggestions on a better place to address the issue, I can move it there. I believe I have eliminated all possibilities that this is due to something on our side, but as always I could be wrong.

**Comment by kakhaUrigashvili** [comment]
@btburton42 would you able to provide more details to help us reproduce the issue on our end? Does it happen only with one skill or all skills? Does this happen if you create a new skill from scratch?

**Comment by btburton42** [comment]
@kakhaUrigashvili here are some helpful details. I should have attached them to the ticket when I created it.

Skill Id: "amzn1.ask.skill.c61c5873-3475-4547-8547-7fec0db280d8"
Session Id: "amzn1.echo-api.session.65c2fe48-e586-4abb-aecc-d14bdd6ffacf"
Request Id: "amzn1.echo-api.request.7303ae1e-9246-46ce-9aa8-6f4bb9471d73"

This behavior can be replicated by attempting a launch request.

It is present in all skills on our account in which I have tested, but I have not yet created a skill from scratch and tested against that.

Also, I have a replay file that I can send you directly if that would help, just let me know where to send it.

Thanks again for checking.

**Comment by kakhaUrigashvili** [comment]
@btburton42 I looked in the logs for request id amzn1.echo-api.request.7303ae1e-9246-46ce-9aa8-6f4bb9471d73 but I did not see anything that could help me to troubleshoot. For example, the response code was 200.

If you could create a skill from scratch, reproduce the issue and describe the steps to reproduce it would very helpful.

**Comment by btburton42** [comment]
@kakhaUrigashvili sure thing. it will take some time to create a new skill, but i will post steps here when done. thanks.

**Comment by btburton42** [comment]
@kakhaUrigashvili I'm still trying to carve out some time to create a skill from scratch to test, but I wanted to pass this over to you in the meantime. When I setup a local proxy, this is the handlerInput I receive from the simulator: 

```
{
  "requestEnvelope": {
    "version": "1.0",
    "session": {
      "new": true,
      "sessionId": "amzn1.echo-api.session.b11ed649-cb24-43c4-8db9-93074cfc5659",
      "application": {
        "applicationId": "amzn1.ask.skill.c61c5873-3475-4547-8547-7fec0db280d8"
      },
      "user": {
        "userId": "[MASKED]"
      }
    },
    "context": {
      "System": {
        "application": {
          "applicationId": "amzn1.ask.skill.c61c5873-3475-4547-8547-7fec0db280d8"
        },
        "user": {
          "userId": "[MASKED]"
        },
        "device": {
          "deviceId": "[MASKED]",
          "supportedInterfaces": {
            "Display": {
              "templateVersion": "1.0",
              "markupVersion": "1.0"
            }
          }
        },
        "apiEndpoint": "https://api.amazonalexa.com",
        "apiAccessToken": "[MASKED]"
      },
      "Display": {}
    },
    "request": {
      "type": "LaunchRequest",
      "requestId": "amzn1.echo-api.request.aedddf69-785e-48ac-9e3d-79527e442681",
      "timestamp": "2020-08-10T16:45:46Z",
      "locale": "en-US",
      "shouldLinkResultBeReturned": false
    }
  },
  "context": {},
  "attributesManager": {},
  "responseBuilder": {}
}
```
I've masked all the relevant details outside of sessionId, requestId, and applicationId.

Per the documentation, (https://developer.amazon.com/en-US/docs/alexa/custom-skills/best-practices-for-designing-skills-for-alexa-enabled-devices-with-a-screen.html#parse-supported-interfaces), a headless device should not include supportedIntefaces.Display object. Is that not correct? 

When I receive handlerInput from a device with a display it also includes a context.Viewport object, but as I understood it, code should look for supportedInterfaces to determine if a display is present or not. 

Does this help you to diagnose?

**Comment by btburton42** [comment]
The ticket issue seems resolved, but it does seem like removing Display from the supportedInterfaces in the simulator is worth looking into, just to conform to spec.

**Comment by kakhaUrigashvili** [comment]
@btburton42 I was able to reach out to person from the team that manages simulate functionality. The issue that you saw was the bug on their end and as you have noticed it has been fixed.

Regarding documentation change about moving Display from the supportedInterfaces in the simulator, I received the explanation that docs do not need to be updated.
The explanation: the device that’s used for simulation is not a headless device, you can think of it as a virtual device. It can support device display if necessary.

I am going to close this issue since there are no action items for the ask cli.

**Comment by btburton42** [comment]
Thanks for looking into this @kakhaUrigashvili and thanks for the detailed explanation.

**Comment by btburton42** [comment]
@kakhaUrigashvili I am seeing what i believe is a related validations error when trying to hit any of our flows that make use of the Speaklist command. Simulator is throwing an error for something that does not error out in the dev console. This flow also used to work previously.

Is there a better way to communicate directly to that simulator team? If so, please let me know. Also, is it better to create a new issue or keep using this one as it seems related?

Posting the masked JSON output below so you can get a better idea about the Speaklist response structure.

Thanks again for your help.

```
{
  "version": "1.0",
  "response": {
    "directives": [
      {
        "type": "Alexa.Presentation.APL.RenderDocument",
        "document": "[MASKED]",
        "datasources": {
          "templateData": {
            "image": {
              "sources": [
                { "url": ["[MASKED]"] }
              ]
            },
            "properties": {
              "hintText": "[MASKED]",
              "listPage": {
                "totalResults": 2,
                "searchQuery": "[MASKED]",
                "listItems": [
                  {
                    "ordinalNumber": 1,
                    "spokenMessageText": {
                      "text": "<speak>[MASKED]</speak>"
                    },
                    "id": "SpeakItemText_0",
                    "touchInfo": "[MASKED]",
                    "listSize": 2,
                    "textContent": {
                      "primaryText": "[MASKED]"
                    },
                    "image": {
                      "sources": [
                        {
                          "url": "[MASKED]"
                        },
                        {
                          "url": "[MASKED]"
                        }
                      ]
                    }
                  },
                  {
                    "ordinalNumber": 2,
                    "spokenMessageText": {
                      "text": "<speak>[MASKED]</speak>"
                    },
                    "id": "SpeakItemText_1",
                    "touchInfo": "[MASKED]",
                    "listSize": 2,
                    "textContent": {
                      "primaryText": "[MASKED]"
                    },
                    "image": {
                      "sources": [
                        {
                          "url": "[MASKED]"
                        },
                        {
                          "url": "[MASKED]"
                        }
                      ]
                    }
                  }
                ]
              }
            },
            "transformers": [
              { "inputPath": "hintText", "transformer": "textToHint" },
              {
                "inputPath": "listPage.listItems[*].spokenMessageText.text",
                "outputName": "textSpeech",
                "transformer": "ssmlToSpeech"
              }
            ],
            "textContent": { "header": { "text": "[MASKED]" } },
            "logoUrl": "[MASKED]",
            "logoLargeUrl": "[MASKED]"
          }
        },
        "token": "multiModalToken"
      },
      {
        "type": "Alexa.Presentation.APL.ExecuteCommands",
        "token": "multiModalToken",
        "commands": [
          {
            "type": "SpeakList",
            "componentId": "SearchSequence",
            "start": 0,
            "count": 1,
            "align": "first"
          }
        ]
      }
    ],
    "outputSpeech": { "type": "SSML", "ssml": "<speak></speak>" },
    "reprompt": {
      "outputSpeech": {
        "type": "SSML",
        "ssml": "<speak>[MASKED]</speak>"
      }
    },
    "shouldEndSession": false
  },
  "userAgent": "ask-node/2.9.0 Node/v12.16.1",
  "sessionAttributes": {}
}
```

**Comment by btburton42** [comment]
Also meant to post the relevant documentation: https://developer.amazon.com/en-US/docs/alexa/alexa-presentation-language/apl-standard-commands.html#speaklist-command

**Comment by kakhaUrigashvili** [comment]
@btburton42 I will check on Monday if there a better way to communicate directly with the simulator team.

**Comment by kakhaUrigashvili** [comment]
@btburton42 i think the issue finally fixed. I did test on my side and it appears to be working correctly. Would you mind verifying on your end?

**Comment by btburton42** [comment]
@kakhaUrigashvili i'm still seeing errors with any response that uses speakList. we've been working with our client rep at amazon in parallel on this issue, who is working in turn with the simulation team. i've shared my dialog replay files with them and they are looking into it. thanks again for getting back to me here.

**Comment by kakhaUrigashvili** [comment]
@btburton42 I will close this issue since it is not related to the ask cli and you have a client rep that is helping you.
